### DISCUSSION TITLE
UC being detected, want to run with a proxy's cert

### DISCUSSION THREAD

**Discussion by dillonhill1** [seleniumbase/SeleniumBase#3075] [Q&A] [answered]
Hello. Any help would be appreciated. I have been using seleniumbase to be able to get to some public data. Seleniumbase's UC mode has been incredibly effective until about a week ago. Now the site I am hitting has begun detecting me. I suspect it Cloudflare based detection. But the site's response is not sending back error messages. It instead just times out on results. If I conduct the same, not using seleniumbase, through Chrome or Firefox it works fine. I have felt limited in using various web proxy services that can get around some of this because they have been detected/blocked. I am wondering, has anyone run seleniumbase with proxy services that use their certificate to make a residential connection/request? I seem to be missing an important step to be able to use wire + seleniumbase to achieve this (such as suggested here: https://github.com/seleniumbase/SeleniumBase/issues/2934). I can't seem to get the desired certificate to be the one used by seleniumbase instead of the OS default certificate? I have attempted to change via code. But when I run certifi.where() it is always just pointing to the default.  My latest attempt to change the default is this:
\
ca_cert_path = '/path/to/cert/here/ca.crt'
context = ssl.create_default_context(cafile=ca_cert_path)

**Comment by mdmintz** [comment, marked as answer]
If you want to set the cert, you'll need to use Wire Mode, which isn't compatible with UC Mode.
But to get past Cloudflare, all you need is UC Mode. Make sure you've read the [UC Mode Docs](https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/uc_mode.md).

The last 3 people who said they had issues with UC Mode had never heard of `uc_gui_click_captcha()`, which is essential to know about for bypassing Cloudflare these days. That, and everything else you need to know for Cloudflare-bypass is covered in the latest UC Mode video tutorial on YouTube: https://www.youtube.com/watch?v=-EpZlhGWo9k.

**Reply by dillonhill1** [reply]
Ok. Much to work with there. I did not use uc_gui_click_captcha(). I will try this.  I am new to this. Thank you! I did miss that in the documentation.

One thing though, the residential IPs appear are still needed for this particular site, so I want to use some proxy services. And given the particular public interest site that I am targeting, I also must use a proxy service's cert. Which does make the cert approach something I would still like to test out. I understand that I would use wire instead of UC with seleniumbase. But does the seleniumbase and wire integration allow you to set something other than the default cert? I can't seem to get that to work.

**Reply by mdmintz** [reply]
For Wire Mode, the same rules apply as in their repo: https://github.com/wkeeling/selenium-wire.
Use `sb.driver` to access the raw driver methods.

**Reply by dillonhill1** [reply]
ok. thank you for all your help.
